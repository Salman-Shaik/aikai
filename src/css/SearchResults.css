.search_results {
  margin-top: 3vh;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
}

.show_items {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 99%;
}

.action_button {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 2vw;
  width: 89%;
  height: 6vh;
  font-size: 1.5em;
  font-weight: bold;
  color: #222222;
  border-radius: 5px;
  background-image: linear-gradient(
    to right bottom,
    #fc5a69,
    #fc676f,
    #fc7275,
    #fc7e7c,
    #fb8883
  );
  background-image: -webkit-linear-gradient(
    to right bottom,
    #fc5a69,
    #fc676f,
    #fc7275,
    #fc7e7c,
    #fb8883
  );
  margin-top: 2vh;
  margin-bottom: 1vh;
}

.minimized {
  transform: rotate(180deg);
}

.show_results {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.search_posters {
  width: 12vw;
  height: 32.1vh;
  border-radius: 3px;
  margin: 3vh 0;
  box-shadow: #1d1d1d 1px 6px 8px 10px !important;
  cursor: pointer;
}

.sectioned_posters {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-evenly;
  width: 99%;
}

.search_text {
  font-size: 1.8em;
  font-weight: bold;
  background-image: -webkit-linear-gradient(-225deg, #e3fdf5 0%, #ffe6fa 100%);
  background-image: linear-gradient(-225deg, #e3fdf5 0%, #ffe6fa 100%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  align-self: flex-start;
  margin-left: 2vw;
  margin-bottom: 3vh;
}

.notice {
  color: #ffefd5;
  font-size: 1.5em;
  font-weight: bold;
  font-style: italic;
}
