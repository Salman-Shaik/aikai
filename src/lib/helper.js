import jwt from "jsonwebtoken";
import _ from 'lodash';
import React from "react";
import {Poster} from "../Components/Homepage/NowPlaying/Poster";
import {genres} from "../genres.json"

const filterPosterPaths = item => {
    return !!item["poster_path"];
};

export const createPosters = (rawJson, setCurrentShowId, setCurrentShowType, setHomePageLoaded) => {
    const filteredList = rawJson.filter(filterPosterPaths);
    return filteredList.map((item, index) => <Poster data={item} key={index} setCurrentShowId={setCurrentShowId}
                                                     setCurrentShowType={setCurrentShowType}
                                                     setHomePageLoaded={setHomePageLoaded}/>);
};

export const refineShowResults = async (results, title) => {
    const flag = await fetchExplicitFlag();
    let movieInfo = results.find(result => {
        const showName = result.title || result.name;
        return !flag ? showName.includes(title) && !result.adult : showName.includes(title);
    });
    return !!movieInfo ? movieInfo : results[0];
};

export const imageUrlBuilder = (imageName) => `https://image.tmdb.org/t/p/original${imageName}`;

export const getGenreNames = (info, showType) => {
    const genre = !!info.genres && info.genres.map(g => g.name);
    const ids = info["genre_ids"];
    if (!_.isEmpty(genre)) return genre.join(", ");
    if (_.isEmpty(ids)) return "";
    const showGenres = genres[showType]
    return ids.map(id => showGenres.find(sg => sg.id === id).name).join(", ");
}

export const getFirstFour = res => {
    return res.slice(0, 4);
};

export const capitalize = s => _.snakeCase(s).split(`_`).map(x => _.capitalize(x)).join(` `);
export const fetchExplicitFlag = () => fetch("/explicitFlag").then(r => r.text()).then(d => JSON.parse(d).flagStatus);

export const handlePerfectShowPromises = (promises, title, setShowInformation, setLoaded, setCurrentShowType, setCurrentShowId, setHomePageLoaded) => {
    let result;
    const findPerfectShow = (val, title, flag) => {
        const name = val.name || val.title;
        return !flag ? name === capitalize(title) && !val.adult : name === capitalize(title);
    };
    const setPerfectShow = values => {
        fetchExplicitFlag().then(flag => {
            result = values.filter(v => !_.isEmpty(v)).find(val => findPerfectShow(val, title, flag));
            if (_.isEmpty(result)) result = values[0];
            setCurrentShowType(values.indexOf(result) === 0 ? "movie" : "tv")
            setShowInformation(result);
            setCurrentShowId((!!result && result.id) || 0);
            setHomePageLoaded(true)
            setLoaded(true);
        })
    };
    Promise.all(promises).then(setPerfectShow);
};

export const getRandomItem = (arr) => {
    const shuffledArr = _.shuffle(arr);
    return shuffledArr[0];
}

export const getCookieValue = (cookieString) => {
    const cookies = document.cookie.split("; ");
    const cookiesObj = {};
    cookies.forEach(cookie => {
        const [key, val] = cookie.split("=");
        cookiesObj[key] = val;
    })
    return cookiesObj[cookieString];
};

export const getJwtToken = (payload) => {
    return jwt.sign(payload, "ADHIIDHIKAADHUADHEIDHI");
}