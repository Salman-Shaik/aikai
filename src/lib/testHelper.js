export const otherShowsMockData = {
  results: [
    {title: "title", posterPath: "/123.png", id: 123},
    {
      title: "title1",
      posterPath: "/456.png",
      id: 456
    },
    {title: "title2", posterPath: "/789.png", id: 789},
    {title: "title3", posterPath: "/012.png", id: 102},
  ],
};

export const showMockData = {
  adult: false,
  backdrop_path: "/x9ezMgOtDPqHatHDvxEG0ILb6Ie.jpg",
  belongs_to_collection: null,
  budget: 61000000,
  genres: [
    {id: 9648, name: "Mystery"},
    {id: 53, name: "Thriller"},
    {id: 18, name: "Drama"},
  ],
  homepage: "http://www.netflix.com/",
  id: 210577,
  imdb_id: "tt2267998",
  original_language: "en",
  original_title: "Gone Girl",
  overview:
    "With his wife's disappearance having become the focus of an intense media circus, a man sees the spotlight turned on him when it's suspected that he may not be innocent.",
  popularity: 44.225,
  poster_path: "/qymaJhucquUwjpb8oiqynMeXnID.jpg",
  release_date: "2014-10-01",
  revenue: 369330363,
  runtime: 149,
  spoken_languages: [{iso_639_1: "en", name: "English"}],
  status: "Released",
  tagline: "You don't know what you've got 'til it's...",
  title: "Gone Girl",
  video: false,
  vote_average: 7.9,
  vote_count: 13273,
};

export const searchResultsMockData = {
  "results":[
    {
      "popularity":750.191,
      "vote_count":255,
      "video":false,
      "poster_path":"\/r4Lm1XKP0VsTgHX4LG4syAwYA2I.jpg",
      "id":590223,
      "adult":false,
      "backdrop_path":"\/lA5fOBqTOQBQ1s9lEYYPmNXoYLi.jpg",
      "original_language":"en",
      "original_title":"Love and Monsters",
      "genre_ids":[
        28,
        12,
        35,
        878
      ],
      "title":"Love and Monsters",
      "vote_average":7.7,
      "overview":"Seven years after the Monsterpocalypse, Joel Dawson, along with the rest of humanity, has been living underground ever since giant creatures took control of the land. After reconnecting over radio with his high school girlfriend Aimee, who is now 80 miles away at a coastal colony, Joel begins to fall for her again. As Joel realizes that there’s nothing left for him underground, he decides against all logic to venture out to Aimee, despite all the dangerous monsters that stand in his way.",
      "release_date":"2020-10-16"
    },
    {
      "popularity":85.464,
      "vote_count":327,
      "video":false,
      "poster_path":"\/6K22JB6fZZLBuM0knfl8rs9Zoxg.jpg",
      "id":644090,
      "adult":false,
      "backdrop_path":"\/i6X9J6AgGhR3FNTtEc5GRs9e3ya.jpg",
      "original_language":"en",
      "original_title":"Love, Guaranteed",
      "genre_ids":[
        10749,
        35
      ],
      "title":"Love, Guaranteed",
      "vote_average":6.6,
      "overview":"A lawyer takes on a new client that wants to sue a dating website because it guarantees love.",
      "release_date":"2020-09-02"
    },
    {
      "popularity":142.787,
      "vote_count":55,
      "video":false,
      "poster_path":"\/vQgXwuJrlpzGDI8169tRQRy71Nv.jpg",
      "id":479259,
      "adult":false,
      "backdrop_path":"\/wUP0KIAXcRevjjJpoSKT7LLhiaK.jpg",
      "original_language":"en",
      "original_title":"Lost Girls & Love Hotels",
      "genre_ids":[
        18
      ],
      "title":"Lost Girls & Love Hotels",
      "vote_average":5.3,
      "overview":"Searching for escape in Tokyo's back alleys, a haunted English teacher explores love and lust with a dashing Yakuza, as their tumultuous affair takes her on a journey through the city's dive bars and three-hour love hotels.",
      "release_date":"2020-09-18"
    },
    {
      "popularity":37.016,
      "vote_count":3402,
      "video":false,
      "poster_path":"\/nMznnEWDAQnUqjOVVen8rLw1MDq.jpg",
      "id":43347,
      "adult":false,
      "backdrop_path":"\/rsBFDOVtsVQ76dcOsRMTmREFffV.jpg",
      "original_language":"en",
      "original_title":"Love & Other Drugs",
      "genre_ids":[
        35,
        18,
        10749
      ],
      "title":"Love & Other Drugs",
      "vote_average":7,
      "overview":"Maggie is an alluring free spirit who won't let anyone - or anything - tie her down. But she meets her match in Jamie, whose relentless and nearly infallible charm serve him well with the ladies and in the cutthroat world of pharmaceutical sales. Maggie and Jamie's evolving relationship takes them both by surprise, as they find themselves under the influence of the ultimate drug: love.",
      "release_date":"2010-11-22"
    },
    {
      "popularity":34.672,
      "id":449176,
      "video":false,
      "vote_count":4538,
      "vote_average":8.2,
      "title":"Love, Simon",
      "release_date":"2018-02-16",
      "original_language":"en",
      "original_title":"Love, Simon",
      "genre_ids":[
        35,
        18,
        10749
      ],
      "backdrop_path":"\/rqMEeajyzSF4PpJtdCB816Iz8Vr.jpg",
      "adult":false,
      "overview":"Everyone deserves a great love story. But for seventeen-year old Simon Spier it's a little more complicated: he's yet to tell his family or friends he's gay and he doesn't know the identity of the anonymous classmate he's fallen for online.",
      "poster_path":"\/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg"
    },
    {
      "popularity":32.241,
      "vote_count":4679,
      "video":false,
      "poster_path":"\/4QNGrKjXS0bxEJgwVhWoHOkbvvV.jpg",
      "id":200727,
      "adult":false,
      "backdrop_path":"\/unJJDP1dE1cfLJojUKVOHJ7zKJe.jpg",
      "original_language":"en",
      "original_title":"Love, Rosie",
      "genre_ids":[
        35,
        10749
      ],
      "title":"Love, Rosie",
      "vote_average":7.8,
      "overview":"Since the moment they met at age 5, Rosie and Alex have been best friends, facing the highs and lows of growing up side by side. A fleeting shared moment, one missed opportunity, and the decisions that follow send their lives in completely different directions. As each navigates the complexities of life, love, and everything in between, they always find their way back to each other - but is it just friendship, or something more?",
      "release_date":"2014-10-16"
    },
    {
      "popularity":33.929,
      "vote_count":24,
      "video":false,
      "poster_path":"\/aZcZKvqZFRxu3TgCiJzFu5Vb1GP.jpg",
      "id":588787,
      "adult":false,
      "backdrop_path":"\/xVr2AyekvqO0DwJg8VwZNaQi5Ji.jpg",
      "original_language":"en",
      "original_title":"Ordinary Love",
      "genre_ids":[
        18,
        10749
      ],
      "title":"Ordinary Love",
      "vote_average":6.5,
      "overview":"An extraordinary look at the lives of a middle-aged couple in the midst of the wife's breast cancer diagnosis.",
      "release_date":"2019-12-06"
    },
    {
      "popularity":52.948,
      "vote_count":4,
      "video":false,
      "poster_path":"\/8p2FKzZE6LA6AIpdOFNICvZDEmB.jpg",
      "id":59315,
      "adult":false,
      "backdrop_path":null,
      "original_language":"en",
      "original_title":"Porno Esotic Love",
      "genre_ids":[

      ],
      "title":"Porno Esotic Love",
      "vote_average":7.3,
      "overview":"Some hardcore's action in this Joe D’ Amato cut and paster that stars Laura Gemser and Dirce Funari as lesbian lovers who work at a kinky nightclub. Gemser dances naked with a python and her junkie girlfriend Funari performs live sex acts both on and off stage (got something on your chin there, sweetie). With insert footage from Eva Nera (aka Black Cobra) and Emanuelle in Bangkok",
      "release_date":"1980-08-20"
    },
    {
      "popularity":38.935,
      "vote_count":462,
      "video":false,
      "poster_path":"\/zn7feouGPU8sELez4qvpp0EtgeQ.jpg",
      "id":624808,
      "adult":false,
      "backdrop_path":"\/ruxOWt2iCPM4tV2PY3OuYJHt4Do.jpg",
      "original_language":"en",
      "original_title":"Love Wedding Repeat",
      "genre_ids":[
        35,
        10749
      ],
      "title":"Love Wedding Repeat",
      "vote_average":5.7,
      "overview":"While trying to make his sister's wedding day go smoothly, Jack finds himself juggling an angry ex-girlfriend, an uninvited guest with a secret, a misplaced sleep sedative, and the girl that got away in alternate versions of the same day.",
      "release_date":"2020-04-10"
    },
    {
      "popularity":22.6,
      "vote_count":5770,
      "video":false,
      "poster_path":"\/wjxIPuCsqmNuAlfQCnP4O5XTXks.jpg",
      "id":50646,
      "adult":false,
      "backdrop_path":"\/ytY39maLlU3Xs2dvSg8PldpIZyT.jpg",
      "original_language":"en",
      "original_title":"Crazy, Stupid, Love.",
      "genre_ids":[
        35,
        18,
        10749
      ],
      "title":"Crazy, Stupid, Love.",
      "vote_average":7.3,
      "overview":"Cal Weaver is living the American dream. He has a good job, a beautiful house, great children and a beautiful wife, named Emily. Cal's seemingly perfect life unravels, however, when he learns that Emily has been unfaithful and wants a divorce. Over 40 and suddenly single, Cal is adrift in the fickle world of dating. Enter, Jacob Palmer, a self-styled player who takes Cal under his wing and teaches him how to be a hit with the ladies.",
      "release_date":"2011-07-29"
    },
    {
      "popularity":21.38,
      "vote_count":4444,
      "video":false,
      "poster_path":"\/7QPeVsr9rcFU9Gl90yg0gTOTpVv.jpg",
      "id":508,
      "adult":false,
      "backdrop_path":"\/dgUJHy2EUBGjWAH6XSl5QHqLXT0.jpg",
      "original_language":"en",
      "original_title":"Love Actually",
      "genre_ids":[
        35,
        18,
        10749
      ],
      "title":"Love Actually",
      "vote_average":7.1,
      "overview":"Follows seemingly unrelated people as their lives begin to intertwine while they fall in – and out – of love. Affections languish and develop as Christmas draws near.",
      "release_date":"2003-09-07"
    },
    {
      "popularity":11.269,
      "vote_count":133,
      "video":false,
      "poster_path":"\/m4FFwMFqDy3LwovqLWBBBaS8pdo.jpg",
      "id":54320,
      "adult":false,
      "backdrop_path":null,
      "original_language":"en",
      "original_title":"Love",
      "genre_ids":[
        18,
        878
      ],
      "title":"Love",
      "vote_average":5.4,
      "overview":"After losing contact with Earth, Astronaut Lee Miller becomes stranded in orbit alone aboard the International Space Station. As time passes and life support systems dwindle, Lee battles to maintain his sanity - and simply stay alive. His world is a claustrophobic and lonely existence, until he makes a strange discovery aboard the ship.",
      "release_date":"2011-02-02"
    }
  ]
}