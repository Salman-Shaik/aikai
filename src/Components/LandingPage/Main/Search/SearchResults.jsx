import _ from "lodash";
import React, { useEffect, useState } from "react";
import "../../../../css/SearchResults.css";
import { getSix, imageUrlBuilder } from "../../../../lib/helper";
import { fetchSearchedShow } from "../../../../lib/showNetworkCalls";
import { Spinner } from "../../../Spinner";

const Section = ({ posters, key }) => (
  <section className="sectioned_posters" key={key}>
    {posters}
  </section>
);

const getSectionedPosters = (map) => {
  const sectionedMap = [];
  for (let i = 0; i < map.length; i += 6)
    sectionedMap.push(<Section posters={getSix(map, i)} key={i} />);
  return sectionedMap;
};

const createSectionedPosters = (
  movieResults,
  setCurrentShowId,
  setShowType,
  setCurrentMenuItem,
  setHomepageLoaded
) => {
  const postersMap = movieResults.map((info, idx) => {
    const id = info.id;
    const title = info.name || info.title;
    const imagePath = info["poster_path"];
    const onClick = () => {
      setCurrentMenuItem("");
      setCurrentShowId(id);
      setShowType();
      setHomepageLoaded(false);
    };
    return (
      <img
        className="search_posters"
        src={imageUrlBuilder(imagePath)}
        alt={title}
        onClick={onClick}
        key={idx}
      />
    );
  });
  return getSectionedPosters(postersMap);
};

export const SearchResults = ({
  currentShowTitle,
  setCurrentShowId,
  setCurrentShowType,
  homepageLoaded,
  setHomePageLoaded,
  setCurrentMenuItem,
}) => {
  const [movieResults, setMovieResults] = useState([]);
  const [tvResults, setTvResults] = useState([]);
  const [movieMinimized, setMovieMinimized] = useState(false);
  const [tvMinimized, setTVMinimized] = useState(true);
  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    fetchSearchedShow("movie", currentShowTitle, setMovieResults);
    fetchSearchedShow(
      "tv",
      currentShowTitle,
      setTvResults,
      setLoaded,
      setHomePageLoaded
    );
  }, [currentShowTitle, setHomePageLoaded]);

  const ResultsComponent = ({
    minimizeMethod,
    minimized,
    anchorText,
    results,
    setShowType,
    setHomepageLoaded,
  }) => {
    return (
      <section className="show_items">
        <section className="action_button" onClick={minimizeMethod}>
          <span className="action_anchor">{anchorText}</span>
          <span className={`action_symbol ${minimized ? "minimized" : ""}`}>
            âŒƒ
          </span>
        </section>
        {!minimized && (
          <section className="show_results">
            {_.isEmpty(results) ? (
              <span className="notice">No Results</span>
            ) : (
              createSectionedPosters(
                results,
                setCurrentShowId,
                setShowType,
                setCurrentMenuItem,
                setHomepageLoaded
              )
            )}
          </section>
        )}
      </section>
    );
  };

  const minimizeMovies = () => {
    setMovieMinimized(!movieMinimized);
    setTVMinimized(true);
  };
  const minimizeTv = () => {
    setTVMinimized(!tvMinimized);
    setMovieMinimized(true);
  };
  const setMovie = () => setCurrentShowType("movie");
  const setTv = () => setCurrentShowType("tv");

  const Component = () => {
    return (
      <section className="search_results">
        <span className="search_text">{`Search Results For ${currentShowTitle}...`}</span>
        <ResultsComponent
          minimizeMethod={minimizeMovies}
          anchorText="Movies"
          results={movieResults}
          setShowType={setMovie}
          minimized={movieMinimized}
          setHomepageLoaded={setHomePageLoaded}
        />
        <ResultsComponent
          minimizeMethod={minimizeTv}
          anchorText="TV Shows"
          results={tvResults}
          setShowType={setTv}
          minimized={tvMinimized}
          setHomepageLoaded={setHomePageLoaded}
        />
      </section>
    );
  };

  return !loaded || !homepageLoaded ? (
    <Spinner loaded={loaded} />
  ) : (
    <Component />
  );
};
