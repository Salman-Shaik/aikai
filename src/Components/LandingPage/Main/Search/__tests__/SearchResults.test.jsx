import { cleanup, fireEvent, render, waitFor } from "@testing-library/react";
import fetchMock from "jest-fetch-mock";
import React from "react";
import { searchResultsMockData } from "../../../../../lib/testHelper";
import { SearchResults } from "../SearchResults";

describe("Search Results", () => {
  let setHomePageLoaded, updateLocation;

  beforeEach(() => {
    fetchMock.doMock();
    setHomePageLoaded = jest.fn();
    updateLocation = jest.fn();
  });

  afterEach(() => {
    cleanup();
    setHomePageLoaded.mockClear();
    updateLocation.mockClear();
  });

  it("Snapshot Test", () => {
    fetch.mockResponse(JSON.stringify(searchResultsMockData), { status: 200 });
    const { container } = render(<SearchResults currentShowTitle="love" />);
    expect(container).toMatchSnapshot();
  });

  it("should maximize movies section and show movie posters by default", async () => {
    fetch.mockResponse(JSON.stringify(searchResultsMockData), { status: 200 });
    const { getByTestId, getAllByTestId } = render(
      <SearchResults
        currentShowTitle="love"
        homepageLoaded={true}
        updateLocation={updateLocation}
      />
    );

    const showResults = await waitFor(() => getByTestId("show_results"));
    const sectionedPosters = await waitFor(() =>
      getAllByTestId("sectioned_posters")
    );

    expect(showResults).toBeDefined();
    expect(sectionedPosters.length).toBe(2);
    expect(sectionedPosters[0].childElementCount).toBe(6);
  });

  it("should maximize tv section and minimize movie section when click on tv bar", async () => {
    fetch.mockResponse(JSON.stringify(searchResultsMockData), { status: 200 });
    const { getAllByTestId } = render(
      <SearchResults
        currentShowTitle="love"
        homepageLoaded={true}
        updateLocation={updateLocation}
        setHomePageLoaded={setHomePageLoaded}
      />
    );

    fireEvent.click(await waitFor(() => getAllByTestId("action_button")[1]));

    const tv = await waitFor(() => getAllByTestId("action_symbol")[1]);
    const movie = await waitFor(() => getAllByTestId("action_symbol")[0]);
    expect(movie.className.includes("minimized")).toBeTruthy();
    expect(tv.className.includes("minimized")).toBeFalsy();
  });

  it("should set setCurrentShowId when clicked on a poster", async () => {
    fetch.mockResponse(JSON.stringify(searchResultsMockData), { status: 200 });
    const { getAllByTestId } = render(
      <SearchResults
        currentShowTitle="love"
        homepageLoaded={true}
        updateLocation={updateLocation}
        setHomePageLoaded={setHomePageLoaded}
      />
    );

    const sectionedPosters = await waitFor(() =>
      getAllByTestId("sectioned_posters")
    );
    fireEvent.click(sectionedPosters[0].firstChild);
    expect(document.cookie).toBe("showId=590223; showType=movie");
  });
});
