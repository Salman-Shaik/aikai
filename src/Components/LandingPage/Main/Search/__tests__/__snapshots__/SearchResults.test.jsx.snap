// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results Snapshot Test 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="spinner"
      role="progressbar"
      style="position: absolute; width: 0px; z-index: 2000000000; left: 50%; top: 50%;"
    >
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-0-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(0deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(0deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-1-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(24deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(24deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-2-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(48deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(48deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-3-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(72deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(72deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-4-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(96deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(96deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-5-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(120deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(120deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-6-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(144deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(144deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-7-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(168deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(168deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-8-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(192deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(192deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-9-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(216deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(216deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-10-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(240deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(240deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-11-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(264deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(264deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-12-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(288deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(288deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-13-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(312deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(312deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
      <div
        style="position: absolute; top: -5px; opacity: 0.25; webkit-animation: opacity-60-25-14-15 1s linear infinite;"
      >
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(0, 0, 0); webkit-box-shadow: 0 0 4px #000; webkit-transform-origin: left; webkit-transform: rotate(336deg) translate(30px,0); webkit-border-radius: 5px; top: 2px;"
        />
        <div
          style="position: absolute; width: 30px; height: 10px; background: rgb(242, 189, 205); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(336deg) translate(30px,0); webkit-border-radius: 5px;"
        />
      </div>
    </div>
  </div>
</div>
`;
